library/ssl_tls13_client.c

```diff
@@ -4104,28 +4118,13 @@ int mbedtls_ssl_handshake_client_step( mbedtls_ssl_context *ssl )
                 mbedtls_ack_add_record( ssl, MBEDTLS_SSL_HS_FINISHED, MBEDTLS_SSL_ACK_RECORDS_RECEIVED );
 #endif /* MBEDTLS_SSL_PROTO_DTLS */
 
-#if defined(MBEDTLS_ZERO_RTT)
-            if( ssl->handshake->early_data == MBEDTLS_SSL_EARLY_DATA_ON )
-            {
-                mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_EARLY_DATA );
-            }
-            else
-#endif /* MBEDTLS_ZERO_RTT */
-            {
-#if defined(MBEDTLS_SSL_TLS13_COMPATIBILITY_MODE)
-                mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_CLIENT_CCS_AFTER_SERVER_FINISHED );
-#else
-                mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_CLIENT_CERTIFICATE );
-#endif /* MBEDTLS_SSL_TLS13_COMPATIBILITY_MODE */
-
-            }
-            break;
+            mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_END_OF_EARLY_DATA );
```

Comment:
Notice that this transition to `MBEDTLS_SSL_END_OF_EARLY_DATA` is without macro guard of `MBEDTLS_ZERO_RTT`. However the case for `MBEDTLS_SSL_END_OF_EARLY_DATA`(L4127) is under `MBEDTLS_ZERO_RTT`.
Does that mean if it may not work if `MBEDTLS_ZERO_RTT` is not defined?

Reply:
That's right, too - I'll fix it.